File: VB/XpoCustomAggregate/DataAccess/DemoDataHelper.vb
Imports DevExpress.Xpo
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace XpoCustomAggregate
	Public Module DemoDataHelper
		Private firstNames() As String = { "Peter", "Ryan", "Richard", "Tom", "Mark"}
		Private lastNames() As String = { "Dolan", "Fischer", "Hamlett", "Hamilton", "Lee", "Lewis" }
		Private productNames() As String = { "Chai", "Chang", "Aniseed Syrup", "Chef Anton's Cajun Seasoning", "Chef Anton's Gumbo Mix", "Grandma's Boysenberry Spread", "Uncle Bob's Organic Dried Pears", "Northwoods Cranberry Sauce", "Mishi Kobe Niku", "Ikura", "Queso Cabrales", "Queso Manchego La Pastora", "Konbu", "Tofu", "Genen Shouyu", "Pavlova", "Alice Mutton"}
		Private Random As New Random(0)

		Public Sub Seed()
			Using uow As New UnitOfWork(XpoDefault.DataLayer)
				If uow.Query(Of Customer)().Count() > 0 Then
					Return
				End If
				Dim customers As New List(Of Customer)()
				For i As Integer = 0 To firstNames.Length - 1
					For j As Integer = 0 To lastNames.Length - 1
						customers.Add(New Customer(uow) With {
							.FirstName = firstNames(i),
							.LastName = lastNames(j)
						})
					Next j
				Next i
				Const recordsCount As Integer = 500
				For i As Integer = 0 To recordsCount - 1
					Dim order As New Order(uow)
					order.ProductName = productNames(Random.Next(productNames.Length))
					order.OrderDate = New Date(Random.Next(2014, 2024), Random.Next(1, 12), Random.Next(1, 28))
					order.Price = 1 + Random.Next(10000) / 100D
					order.Quantity = 1 + Random.Next(100)
					order.Customer = customers(Random.Next(customers.Count))
				Next i
				uow.CommitChanges()
			End Using
		End Sub
	End Module
End Namespace

File: CS/XpoCustomAggregate/DataAccess/DemoDataHelper.cs
using DevExpress.Xpo;
using System;
using System.Collections.Generic;
using System.Linq;

namespace XpoCustomAggregate {
    public static class DemoDataHelper {
        private static string[] firstNames = new string[] {
            "Peter", "Ryan", "Richard", "Tom", "Mark"};
        private static string[] lastNames = new string[] {
            "Dolan", "Fischer", "Hamlett", "Hamilton", "Lee", "Lewis" };
        private static string[] productNames = new string[] {
            "Chai", "Chang", "Aniseed Syrup", "Chef Anton's Cajun Seasoning",
            "Chef Anton's Gumbo Mix", "Grandma's Boysenberry Spread",
            "Uncle Bob's Organic Dried Pears", "Northwoods Cranberry Sauce",
            "Mishi Kobe Niku", "Ikura", "Queso Cabrales", "Queso Manchego La Pastora",
            "Konbu", "Tofu", "Genen Shouyu", "Pavlova", "Alice Mutton"};
        private static Random Random = new Random(0);

        public static void Seed() {
            using(UnitOfWork uow = new UnitOfWork(XpoDefault.DataLayer)) {
                if(uow.Query<Customer>().Count() > 0) {
                    return;
                }
                List<Customer> customers = new List<Customer>();
                for(int i = 0; i < firstNames.Length; i++) {
                    for(int j = 0; j < lastNames.Length; j++) {
                        customers.Add(new Customer(uow) {
                            FirstName = firstNames[i],
                            LastName = lastNames[j]
                        });
                    }
                }
                const int recordsCount = 500;
                for(int i = 0; i < recordsCount; i++) {
                    Order order = new Order(uow);
                    order.ProductName = productNames[Random.Next(productNames.Length)];
                    order.OrderDate = new DateTime(Random.Next(2014, 2024), Random.Next(1, 12), Random.Next(1, 28));
                    order.Price = 1 + Random.Next(10000) / 100m;
                    order.Quantity = 1 + Random.Next(100);
                    order.Customer = customers[Random.Next(customers.Count)];
                }
                uow.CommitChanges();
            }
        }
    }
}